core/runtime.rs

```diff
@@ -605,9 +605,10 @@ impl JsRuntime {
       None
     };
 
-    let loader = options
-      .module_loader
-      .unwrap_or_else(|| Rc::new(NoopModuleLoader));
+    let loader = Rc::new(crate::modules::InternalModuleLoader::new(
+      options.module_loader,
+    ));
```

Comment:
We should rework this - we shouldn't need `InternalModuleLoader` in every single case.

Reply:
To that end, we should add a `Mode` enum to `JsRuntime`, that specifies if:
a) we're running without a snapshot
b) we're gonna build a snapshot
c) we're gonna build a snapshot from existing snapshot
d) we're running with a snapshot and not gonna build another one

Reply:
I guess this is actually covered by the `SnapshotOptions` enum